src/test/java/tracko/model/person/NameContainsKeywordsPredicateTest.java

```diff
@@ -1,76 +1,78 @@
+//        List<String> firstPredicateKeywordList = Collections.singletonList("first");
+//        List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
+//
+//        NameContainsKeywordsPredicate firstPredicate = new NameContainsKeywordsPredicate(firstPredicateKeywordList);
+//        NameContainsKeywordsPredicate secondPredicate = new NameContainsKeywordsPredicate(secondPredicateKeywordList);
+//
+//        // same object -> returns true
+//        assertTrue(firstPredicate.equals(firstPredicate));
+//
+//        // same values -> returns true
+//        NameContainsKeywordsPredicate firstPredicateCopy =
+//        new NameContainsKeywordsPredicate(firstPredicateKeywordList);
+//        assertTrue(firstPredicate.equals(firstPredicateCopy));
+//
+//        // different types -> returns false
+//        assertFalse(firstPredicate.equals(1));
+//
+//        // null -> returns false
+//        assertFalse(firstPredicate.equals(null));
+//
+//        // different person -> returns false
+//        assertFalse(firstPredicate.equals(secondPredicate));
+//    }
+//
+//    @Test
+//    public void test_nameContainsKeywords_returnsTrue() {
+//        // One keyword
+//        NameContainsKeywordsPredicate predicate =
+//        new NameContainsKeywordsPredicate(Collections.singletonList("Alice"));
+//        assertTrue(predicate.test(new OrderBuilder().withName("Alice Bob").build()));
+//
+//        // Multiple keywords
+//        predicate = new NameContainsKeywordsPredicate(Arrays.asList("Alice", "Bob"));
+//        assertTrue(predicate.test(new OrderBuilder().withName("Alice Bob").build()));
+//
+//        // Only one matching keyword
+//        predicate = new NameContainsKeywordsPredicate(Arrays.asList("Bob", "Carol"));
+//        assertTrue(predicate.test(new OrderBuilder().withName("Alice Carol").build()));
+//
+//        // Mixed-case keywords
+//        predicate = new NameContainsKeywordsPredicate(Arrays.asList("aLIce", "bOB"));
+//        assertTrue(predicate.test(new OrderBuilder().withName("Alice Bob").build()));
+//    }
+//
+//    @Test
+//    public void test_nameDoesNotContainKeywords_returnsFalse() {
+//        // Zero keywords
+//        NameContainsKeywordsPredicate predicate = new NameContainsKeywordsPredicate(Collections.emptyList());
+//        assertFalse(predicate.test(new OrderBuilder().withName("Alice").build()));
+//
+//        // Non-matching keyword
+//        predicate = new NameContainsKeywordsPredicate(Arrays.asList("Carol"));
+//        assertFalse(predicate.test(new OrderBuilder().withName("Alice Bob").build()));
+//
+//        // Keywords match phone, email and address, but does not match name
+//        predicate = new NameContainsKeywordsPredicate(Arrays.asList("12345", "alice@email.com", "Main", "Street"));
+//        assertFalse(predicate.test(new OrderBuilder().withName("Alice").withPhone("12345")
+//                .withEmail("alice@email.com").withAddress("Main Street").build()));
+//    }
```

Comment:
Can maybe look into re-implementing these test cases somehow. I think a lot of them can be reimplemented by changing a few lines here and there.

Reply:
yup, reimplemented the tests